libraries/@mattermost/rnshare/android/src/main/java/com/mattermost/rnshare/ShareWorker.kt

```diff
@@ -22,11 +24,60 @@ import org.json.JSONException
 import org.json.JSONObject
 import java.io.File
 import java.io.IOException
+import java.io.InputStream
+import java.security.MessageDigest
+import java.security.cert.CertificateFactory
+import java.security.cert.X509Certificate
 import java.util.Objects
 
-class ShareWorker(context: Context, workerParameters: WorkerParameters) : Worker(context, workerParameters) {
-    private val okHttpClient = OkHttpClient()
+class ShareWorker(private val context: Context, workerParameters: WorkerParameters) : Worker(context, workerParameters) {
     private val jsonType: MediaType? = "application/json; charset=utf-8".toMediaTypeOrNull()
+    private val okHttpClient: OkHttpClient
+        get() {
+            val builder = OkHttpClient.Builder()
+            val fingerprintsMap = getCertificatesFingerPrints()
+            if (fingerprintsMap.isNotEmpty()) {
+                val pinner = CertificatePinner.Builder()
+                for ((domain, fingerprints) in fingerprintsMap) {
+                    for (fingerprint in fingerprints) {
+                        pinner.add(domain, "sha256/$fingerprint")
+                    }
+                }
+                val certificatePinner = pinner.build()
+                builder.certificatePinner(certificatePinner)
+            }
+            return builder.build()
+        }
+
+    private fun getCertificateFingerPrint(certInputStream: InputStream): String {
+        val certFactory = CertificateFactory.getInstance("X.509")
+        val certificate = certFactory.generateCertificate(certInputStream) as X509Certificate
+        val sha256 = MessageDigest.getInstance("SHA-256")
+        Base64.encodeToString(sha256.digest(certificate.publicKey.encoded), Base64.NO_WRAP)
+        val fingerprintBytes = sha256.digest(certificate.publicKey.encoded)
+        return Base64.encodeToString(fingerprintBytes, Base64.NO_WRAP)
+    }
+
+    private fun getCertificatesFingerPrints(): Map<String, List<String>> {
```

Comment:
Is there a way to share the code with the network client?

Reply:
There is if we add the network client as a dependency of this one, so in the end even if the code is dupe is less expensive this way

Reply:
I was going to ask the same thing - especially considering I left comments on this code in the network client.